ci/jenkins/test.sh

```diff
@@ -38,6 +38,7 @@ IP_MODE=""
 K8S_VERSION="1.28.2-00"
 WINDOWS_YAML_NAME="antrea-windows"
 WIN_IMAGE_NODE=""
+FLEXIBLE_IPAM=false
```

Comment:
May I know why `FLEXIBLE_IPAM` is needed? Can we just use `TESTBED_TYPE` to determine the kind ipam test?

Reply:
Because in test.sh before running actual test it will run clean_antrea function and there I have added  if condition to delete added ip routes for non-ipam pods, if there are no routes then it will show the error:

if [[ $TESTBED_TYPE == "kind-flexible-ipam" && $FLEXIBLE_IPAM == "true" ]]; then

 cidrs=($(kubectl get nodes -o jsonpath='{.items[*].spec.podCIDR}')) 
 for cidr in "${cidrs[@]}"; do 
   sudo ip route del $cidr 
 done
fi


that's why I am setting FLEXIBLE_IPAM = true if I am adding the routes:

if [[ $TESTBED_TYPE == "kind-flexible-ipam" ]]; then
                  kubectl get nodes -o wide --no-headers=true |  awk '{print $1}' | while read nodes; do
                    configure_kind_ipam_routes "${nodes}"
                  done
                  FLEXIBLE_IPAM=true
  fi

Reply:
If `FLEXIBLE_IPAM` is only used to determine whether to del routes, we can directly check whether the corresponding routes have been added. In some corner cases, when the test is interrupted during the process of adding routes, there will still be residual routes that will not be deleted.

e.g. 
if [[ $TESTBED_TYPE == "kind-flexible-ipam" ]]; then
    cidrs=($(kubectl get nodes -o jsonpath='{.items[*].spec.podCIDR}')) 
    for cidr in "${cidrs[@]}"; do 
      if ip route show | grep -q $cidr; then
        sudo ip route del $cidr 
      fi
    done
  fi